sys/net/include/ieee802154.h

```diff
@@ -0,0 +1,211 @@
+ * @{
+ */
+#define IEEE_802154_SHORT_MCAST_ADDR    (0xffff)
+#define IEEE_802154_LONG_MCAST_ADDR     {0xff, 0xff, 0xff, 0xff, \
+                                          0xff, 0xff, 0xff, 0xff}
+/** @} */
+
+/**
+ * @brief The default PAN id to use if not specified otherwise by upper layers
+ */
+#define IEEE_802154_DEFAULT_PAN_ID      (1)
+
+/**
+ * @brief   Transform 16-bit number from network order (big-endian) to
+ *          little-endian byte order (as used by IEEE 802.15.4).
+ *
+ * @NOTE    TODO: remove and use macros from /core/include/byteorder.h
+ */
+#define NTOLES(a)   (((a) >> 8) | (((a) & 0x00ff) << 8))
+
+/**
+ * @brief   Transform 16-bit number from little-endian byte order to network
+ *          order (big-endian).
+ *
+ * @NOTE    TODO: remove and use macros from /core/include/byteorder.h
+ */
+#define LETONS(a)   NTOLES(a)
+
+/**
+ * @brief   Transform 16-bit number from host byte order to little-endian byte
+ *          order (as used by IEEE 802.15.4).
+ *
+ * @NOTE    TODO: remove and use macros from /core/include/byteorder.h
+ */
+#define HTOLES(a)   a
+
+/**
+ * @brief   Transform 16-bit number from little-endian byte order to host byte
+ *          order.
+ *
+ * @NOTE    TODO: remove and use macros from /core/include/byteorder.h
+ */
+#define LETOHS(a)   HTOLES(a)
+
+/**
+ * @brief   IEEE802.15.4 frame descriptor
+ */
+typedef struct __attribute__((packed)) {
+    uint16_t fcf;               /**< frame control field */
+    uint8_t seq_nr;             /**< sequence number */
+    uint16_t dest_pan_id;       /**< destination PAN id */
+    uint8_t dest_addr[8];       /**< destination address */
+    uint16_t src_pan_id;        /**< source PAN id */
+    uint8_t src_addr[8];        /**< source address */
+    uint8_t *payload;           /**< payload pointer */
+    uint8_t payload_len;        /**< size of payload in byte */
+} ieee802154_frame_t;
+
+/**
+ * Structure to represent an IEEE 802.15.4 packet for the transceiver.
+ */
+typedef struct __attribute__(( packed )) {
+    /* @{ */
+    uint8_t processing;         /** < internal processing state */
+    uint8_t length;             /** < the length of the frame of the frame including fcs*/
+    ieee802154_frame_t frame;   /** < the ieee802154 frame */
+    int8_t rssi;                /** < the rssi value */
+    uint8_t crc;                /** < 1 if crc was successfull, 0 otherwise */
+    uint8_t lqi;                /** < the link quality indicator */
+    /* @} */
+} ieee802154_packet_t;
+
+/**
+ * @brief   Convert the given frame data-structure to raw binary data
+ *
+ * @note    The function asserts that the given buffer is large enough to hold
+ *          the raw header
+ *
+ * @param[in]  frame            frame to convert
+ * @param[out] buf              memory to save raw frame to
+ *
+ * @return                      the actual size of the raw header in byte
+ */
+uint8_t ieee802154_frame_init(ieee802154_frame_t *frame, uint8_t *buf);
```

Comment:
normally yes, but this part I didn't touch - so room for improvement in the next PR...